src/main/java/com/isroot/stash/plugin/YaccServiceImpl.java

```diff
@@ -123,6 +126,21 @@ private boolean isCommitExcluded(Settings settings, YaccCommit commit) {
 
         return false;
     }
+    
+    private boolean isBranchExcluded(Settings settings, String branchName) {
+        // Exclude by Regex setting
+        String excludeBranchRegex = settings.getString("excludeBranchRegex");
+
+        if(excludeBranchRegex != null && !excludeBranchRegex.isEmpty()) {
+            Pattern pattern = Pattern.compile(excludeBranchRegex);
+            Matcher matcher = pattern.matcher(branchName);
+            if(matcher.find()) {
```

Comment:
This should use `matches`. `find` makes it easy to accidentally exclude the wrong branch due to partial matches. Rest of PR looks fine so I can fix this after merge.